scripts/buildorder.py

```diff
@@ -385,16 +437,25 @@ def main():
             packages_directories.insert(0, os.path.dirname(package))
     pkgs_map = read_packages_from_directories(packages_directories, fast_build_mode, full_buildorder)
 
+    if get_list_cirdep:
+        pkgname = None
+        if not full_buildorder:
+            pkgname = package.split("/")[-1]
+            if "gpkg" in package.split("/")[-2].split("-") and not has_prefix_glibc(pkgname):
+                pkgname += "-glibc"
+        get_list_cyclic_dependencies(pkgs_map, pkgname=pkgname, build_mode=not(fast_build_mode))
```

Comment:
Previously, the `-l` flag scanned all packages with all dependencies, which took a lot of time. But I found solutions to reduce packages and dependencies so that scanning is faster.

Also, note that only with the `-l` flag the scanning occurs without taking into subpackages. To take into subpackages, simply add the `-i` flag.